crates/abq_cli/src/main.rs

```diff
@@ -654,95 +673,69 @@ impl QueueLocation {
+                )),
             }
         }
-        None => match config.queue_addr {
-            Some(queue_addr) => QueueLocation::Remote(queue_addr),
-            None => QueueLocation::Ephemeral {
-                opt_tls_key: config.tls_key,
-            },
-        },
     }
 }
 
-async fn resolve_config(
+struct ResolveConfigOptions {
     token_from_cli: Option<UserToken>,
     queue_addr_from_cli: Option<SocketAddr>,
     tls_cert_from_cli: Option<Vec<u8>>,
     tls_key: Option<Vec<u8>>,
-    access_token: &Option<AccessToken>,
-    run_id: &RunId,
+    api_config: Option<ConfigFromApi>,
     explicit_run_id_provided: bool,
-) -> anyhow::Result<ResolvedConfig> {
-    let (
-        queue_addr_from_api,
-        token_from_api,
-        tls_from_api,
-        usage_error_from_api,
-        rwx_access_token_kind,
-    ) = match access_token.as_ref() {
-        Some(access_token) => match get_config_from_api(access_token, run_id).await? {
-            ConfigFromApi::Success(config) => (
-                Some(config.queue_addr),
-                Some(config.token),
-                config.tls_public_certificate,
-                None,
-                Some(config.rwx_access_token_kind),
-            ),
-            ConfigFromApi::Unsupported(config) => (
-                None,
-                None,
-                None,
-                Some(config.usage_error),
-                Some(config.rwx_access_token_kind),
-            ),
-        },
-        None => (None, None, None, None, None),
-    };
-
-    let token = token_from_api.or(token_from_cli);
-    let queue_addr = queue_addr_from_api.or(queue_addr_from_cli);
-    let tls_cert = tls_from_api.or(tls_cert_from_cli);
+}
 
-    let queue_location_config = QueueLocationConfig {
-        access_token_kind: rwx_access_token_kind,
-        run_id_provided: explicit_run_id_provided,
-        queue_addr,
-        usage_error_from_api,
+fn resolve_config_with_api(options: ResolveConfigOptions) -> ResolvedConfig {
+    let ResolveConfigOptions {
+        token_from_cli,
+        queue_addr_from_cli,
+        tls_cert_from_cli,
         tls_key,
-    };
-    let queue_location = determine_queue_location(queue_location_config);
+        api_config,
+        explicit_run_id_provided,
+    } = options;
 
-    Ok(ResolvedConfig {
-        token,
-        tls_cert,
-        rwx_access_token_kind,
+    if let Some(api_config) = api_config {
+        if api_config.access_token_kind() != AccessTokenKind::Personal || explicit_run_id_provided {
+            return resolve_api_config(api_config);
+        }
+    }
+
+    let queue_location = match queue_addr_from_cli {
+        Some(addr) => QueueLocation::Remote(addr),
+        None => QueueLocation::Ephemeral {
+            opt_tls_key: tls_key,
+        },
+    };
+    ResolvedConfig {
+        token: token_from_cli,
+        tls_cert: tls_cert_from_cli,
+        rwx_access_token_kind: None,
         queue_location,
-    })
+    }
```

Comment:
If I'm following it goes:

1. `get_config_from_api`
2. `resolve_config_with_api`
3a. arrive at `ResolvedConfig` from calling `resolve_api_config`
3b. arrive at `ResolvedConfig` inline after determining queue location

Total arm chair rust critique, and this is absolutely already a huge improvement, so please don't take this the wrong way:

I find a little confusing is that `resolve_api_config` seems like a thing you can do generally, but it's actually very important it's only called from `resolve_config_with_api` if I'm correct in my understanding.

Also, `resolve_config_with_api` reads to me as if you're using the remote api to resolve the config.

```
fetch -> resolve -> resolve variant a
                \-> resolve variant b
```